### ISSUE TITLE
Jest 28 + firebase compat: SyntaxError: Cannot use import statement outside a module

### ISSUE THREAD

**Issue by LanderBeeuwsaert** [firebase/firebase-js-sdk#6351]
### [REQUIRED] Describe your environment

  * Operating System version: Windows 10
  * Browser version: /
  * Firebase SDK version: 9.8.3
  * Firebase Product: any

### [REQUIRED] Describe the problem

We have an angular + angularFire + firebase + jest application
while upgrading to angular 14, we were required to also upgrade jest-preset-angular to 12.1.0 and jest to 28.1.1.
After doing this, we have the error you can find below, for all our jest tests.
The most relevant part seems to be: "SyntaxError: Cannot use import statement outside a module"
Which happens in the compatibility code, which I guess is not using ESM?
So I'm at loss on how to solve this, and what is actually the correct repo to post this. (firebase, jest, jest-preset-angular).
Any ideas?

*********************************************
 spec/tests/app/import-service/validators/length.validator.spec.ts
  ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is
 not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.      

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config optio
n.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    C:\lander\devprojects\acro-companion\node_modules\firebase\compat\app\dist\index.esm.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import firebase from '@firebase/app-compat';       
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      1 | import { CellMetaData, VALIDATOR_RESULT_TYPE, CellRawValue, RawSpreadSheet, ValidatorResult, ValidatorMetaData } from '../../sp
readsheet-config.model';
      2 | import { DataValidator } from '../data-validator.abstract';
    > 3 | import firebase from 'firebase/compat/app';
        | ^
      4 | import firestore = firebase.firestore;
      5 | import { DATA_VALIDATOR_ID } from '../../data-validators.enum';
      6 |

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1796:14)
      at Object.<anonymous> (src/app/core/services/generic-incoming-data-process/validators/firestore-doc-validators/firestore-doc-valida
tor.ts:3:1)

 FAIL  spec/tests/app/import-service/validators/trimmed.validator.spec.ts
  ● Test suite failed to run
******************************************

**Comment by hsubox76** [comment]
I'm not sure, I think it might be something in the Jest or jest-preset-angular config. The firebase/app/compat package is an ES module and always has been, and we should have the Node exports fields set up correctly in package.json as well. I searched around for "jest" and that specific error and there are a lot of suggested solutions (mostly involving transpilation), but they're all different so I don't want to pick one out and possibly point you in the wrong direction. I think it might be worth looking at some of these and seeing if any of them apply.

**Comment by LanderBeeuwsaert** [comment]
Hi @hsubox76 thank you for the answer.
I've tried quite a few of the proposed solutions I could find but I haven't been able to get it working.
What I did in the end was to make a new project to see if I can reproduce it with a minimal setup, and yes, just one very small test gives the same issue.
The repro is here:
https://github.com/LanderBeeuwsaert/clean-ng-project
Any chance that somebody from the firebase team could have a look?
